### PR TITLE
Add metrics scrape configuration for Istio & envoy

### LINKED ISSUE #306: Configure the local setup to scrape metrics from Istio gateways in each spoke cluster
*What*

Building on #304, configure Istio metrics scraping from the prometheus instance(s) in the spoke clusters.
Further info on how to scrape metrics from istio is available at https://istio.io/latest/docs/ops/integrations/prometheus/

This task can be seen as complete if a response can be retrieved from the thanos query API in the hub for an istio metric that's federated from 2 clusters.

e.g. 
Rate of aggregate requests per workload across all clusters
```
sum(rate(istio_requests_total{}[5m])) by(destination_workload)
```

Rate of aggregate requests per pod (per cluster, if 1 pod per cluster)
```
sum(rate(istio_requests_total{}[5m])) by(pod)
```

### PR DESCRIPTION
**Pull request by david-martin**
Closes #306 

Prometheus scrape configuration is added so that metrics from Istio and envoy are scraped from all clusters (including the hub, as that may be also a cluster where workloads run).

All metrics are remote written to thanos in the hub. This is handled by the existing setup.

The metrics federation walkthrough has been updated with a couple of queries that leverage istio request metrics.

To verify, follow through the updated metrics federation walkthrough.

### CODE REVIEW

**Review by maleck13 — COMMENTED** [review]
These changes look reasonable to me, will give it a try also

File: docs/how-to/metrics-federation.md [hunk 1]
```diff
@@ -41,3 +41,33 @@ You should see a response in the table view.
...
 <img src="images/metrics-federation-example-data.png" alt="architecture" width="600"/>
+
+To query Istio workload metrics, you should first deploy a Gateway & HttpRoute, and send traffic to it.
+The easiest way to do this is by following the steps in the [OCM Walkthrough](./ocm-control-plane-walkthrough.md) (without re-running the `make local-setup` step, as that should have already been run with the `METRICS_FEDERATION` flag above).
+
+Use `curl` to send some traffic to the application
+
+```bash
+while true; do curl -k https://$MGC_SUB_DOMAIN && sleep 5;
+```
```
Comment (eoinfennessy) [inline docs/how-to/metrics-federation.md:52]:
```suggestion
```bash
while true; do curl -k https://$MGC_SUB_DOMAIN && sleep 5; done
```
```

File: docs/how-to/metrics-federation.md [hunk 2]
```diff
@@ -41,3 +41,32 @@ You should see a response in the table view.
...
+Use `curl` to send some traffic to the application
+
+```bash
+while true; do curl -k https://$MGC_SUB_DOMAIN && sleep 5; done
+
+Open the Thanos Query UI again and try the below query:
+
+```
+sum(rate(istio_requests_total{}[5m])) by(destination_workload)
+```
```
Comment (eoinfennessy) [inline docs/how-to/metrics-federation.md:57]: @david-martin, I'm getting "empty query result" when running this and the other Istio query. I've completed the OCM walkthrough and have been curl-ing `https://$MGC_SUB_DOMAIN` for about half an hour now and getting responses similar to the following.
```
Host: thanos-test.ef.hcpapps.net
Accept: */*
User-Agent: curl/8.0.1
X-B3-Sampled: 0
X-B3-Spanid: 23529e261c54b29d
X-B3-Traceid: 82e20dc220ace15823529e261c54b29d
X-Envoy-Attempt-Count: 1
X-Envoy-Decorator-Operation: echo.default.svc.cluster.local:8080/*
X-Envoy-Internal: true
X-Envoy-Peer-Metadata: ChQKDkFQUF9DT05UQUlORVJTEgIaAAoaCgpDTFVTVEVSX0lEEgwaCkt1YmVybmV0ZXMKHAoMSU5TVEFOQ0VfSVBTEgwaCjEwLjI0LjAuNjUKGQoNSVNUSU9fVkVSU0lPThIIGgYxLjE3LjAKtgIKBkxBQkVMUxKrAiqoAgo+CixjbHVzdGVyLm9wZW4tY2x1c3Rlci1tYW5hZ2VtZW50LmlvL3BsYWNlbWVudBIOGgxodHRwLWdhdGV3YXkKIwoVaXN0aW8uaW8vZ2F0ZXdheS1uYW1lEgoaCHByb2Qtd2ViChkKDGlzdGlvLmlvL3JldhIJGgdkZWZhdWx0Ch0KE2t1YWRyYW50LmlvL21hbmFnZWQSBhoEdHJ1ZQozCh9zZXJ2aWNlLmlzdGlvLmlvL2Nhbm9uaWNhbC1uYW1lEhAaDnByb2Qtd2ViLWlzdGlvCi8KI3NlcnZpY2UuaXN0aW8uaW8vY2Fub25pY2FsLXJldmlzaW9uEggaBmxhdGVzdAohChdzaWRlY2FyLmlzdGlvLmlvL2luamVjdBIGGgR0cnVlChoKB01FU0hfSUQSDxoNY2x1c3Rlci5sb2NhbAopCgROQU1FEiEaH3Byb2Qtd2ViLWlzdGlvLTc1YjdjYzQ4NDUtOWd6bWMKLgoJTkFNRVNQQUNFEiEaH2t1YWRyYW50LW11bHRpLWNsdXN0ZXItZ2F0ZXdheXMKagoFT1dORVISYRpfa3ViZXJuZXRlczovL2FwaXMvYXBwcy92MS9uYW1lc3BhY2VzL2t1YWRyYW50LW11bHRpLWNsdXN0ZXItZ2F0ZXdheXMvZGVwbG95bWVudHMvcHJvZC13ZWItaXN0aW8KFwoRUExBVEZPUk1fTUVUQURBVEESAioACiEKDVdPUktMT0FEX05BTUUSEBoOcHJvZC13ZWItaXN0aW8=
X-Envoy-Peer-Metadata-Id: router~10.24.0.65~prod-web-istio-75b7cc4845-9gzmc.kuadrant-multi-cluster-gateways~kuadrant-multi-cluster-gateways.svc.cluster.local
X-Forwarded-For: 10.24.0.1
X-Forwarded-Proto: https
X-Request-Id: 7d0a829f-72d1-4155-9467-1be0762b1541
```
All previous queries are working as described. Any idea why these Istio-related ones aren't  working for me?

Comment (eoinfennessy) [reply docs/how-to/metrics-federation.md:57]: Apologies Dave - this was a silly mistake on my part. I had accidentally set the graph's "end time" in the Thanos dashboard to a fixed time before I started the OCM walkthrough. All is working as expected now.

**Review by maleck13 — APPROVED** [review]
changes look fine and walkthrough worked as expected

**Pull request merged**